src/main/java/org/spongepowered/api/command/CommandCallable.java

```diff
@@ -60,10 +64,13 @@
      *
      * @param source The command source
      * @param arguments The arguments entered up to this point
+     * @param location The {@link Location} being looked at by the user,
+     * if available
+     *
      * @return A list of suggestions
      * @throws CommandException Thrown if there was a parsing error
      */
-    List<String> getSuggestions(CommandSource source, String arguments) throws CommandException;
+    List<String> getSuggestions(CommandSource source, String arguments, @Nullable Location<World> location) throws CommandException;
```

Comment:
I don't think this fits well in the API. We already have the blockray system which is imo a better way (even though nms implements this way). Currently, iirc the way this is done is the client tells the server what it's looking at and the server just trusts the client -- we don't need to do this since we can raytrace -- and plugins that raytrace can handle that.

This also isn't good for any source that doesn't have a looking-at -- if it's kept it should be an Optional -- though again imo it is too player-specific.